lib/src/rules/noop_primitive_operations.dart

```diff
@@ -66,10 +70,21 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
   }
 
+  @override
+  void visitInterpolationExpression(InterpolationExpression node) {
+    _checkToStringInvocation(node.expression);
+  }
+
   @override
   void visitMethodInvocation(MethodInvocation node) {
     var type = node.realTarget?.staticType;
-    if (type == null) return;
+    if (type == null) {
+      // print(xxx.toString())
+      if (node.methodName.staticElement?.isDartCorePrint ?? false) {
+        _checkToStringInvocation(node.argumentList.arguments.first);
```

Comment:
The code under analysis might be incomplete, and have `print();` without any arguments.